ruby_on_rails/testing/unit_testing.md

```diff
@@ -0,0 +1,84 @@
+### Introduction
+
+You may remember testing from the Ruby section and the emphasis on the importance of testing. Similar to that section, testing in Ruby on Rails is important to ensure your application is stable and to catch bugs before you expand your application even more! The first kind of testing that we will cover will be unit tests.
+
+### Lesson Overview
+
+This lesson will cover a few points.
+
+*   What are unit tests?
+*   Why are unit tests important?
+*   Unit tests in the context of Ruby on Rails.
+
+## What are unit tests?
+
+A unit test is testing a single unit of code to ensure that method/unit works as it is supposed to! In Ruby on Rails you will also hear them reffered to as model tests. These model tests specifically are used to test the logic behind your objects. This can include validating user logins, user form submissions and other logic between models. Such as if you need to ensure your object method for generating discounts work properly. It's important as you scale your projects to ensure that the logic behind your models works as your project consistently expands. This helps make the project more maintainable as bugs will be caught earlier rather than when your project gets larger and those bugs become harder to to debug!
+
+Earlier you used RSpec for testing projects in the Ruby section. While Ruby on Rails comes with it's own testing framework. RSpec is a very popular framework for testing even in Ruby on Rail's. As such we will be using RSpec for the Ruby on Rails testing section. 
+
+## Setting up RSpec
+
+Installing RSpec for rails is a relatively straightforward task.
+
+First we add it the development and test group of the gemfile. Navigate to the gemfile and add this to the development/test section.
+
+~~~
+
+gem 'rspec-rails', '~> 6.0.0'
+
+~~~
+
+Next in your terminal run the following two commands separately.
+
+~~~
+
+bundle install
+
+~~~
+
+~~~
+
+rails generate rspec:install
+
+~~~
+
+Now the appropriate files should have been installed.
```

Comment:
I wonder if it would be helpful to refer learners to follow the installation instructions that are included in the readme on github? That way they get more and more used to learning straight from the source, and we have less to keep updated when things change?

https://github.com/rspec/rspec-rails#installation